mac_internet_sharing/__main__.py

```diff
@@ -17,6 +18,67 @@
 logger = logging.getLogger(__name__)
 
 
+async def plug_n_share_task(network_service_name: Optional[str], timeout: int = 5):
+    """ Continuously monitor for USB device changes and update SharingDevices. """
+    network_service = get_network_service(network_service_name)
+
+    logger.info('Starting USB device monitoring...')
+    prev_usb_devices = get_apple_usb_ethernet_interfaces()
+    logger.info(f'Connected devices: {set(prev_usb_devices.keys())}')
+
+    # Configure sharing with the initially detected interfaces.
+    configure(network_service, list(prev_usb_devices.values()))
+    await set_sharing_state(SharingState.ON)
+    while True:
+        current_usb_devices = get_apple_usb_ethernet_interfaces()
+        if prev_usb_devices != current_usb_devices:
+            if len(prev_usb_devices) > len(current_usb_devices):
+                logger.info(f'Removing devices {set(prev_usb_devices.keys()) - set(current_usb_devices.keys())}')
+            else:
+                logger.info(f'Adding devices {set(current_usb_devices.keys()) - set(prev_usb_devices.keys())}')
+            prev_usb_devices = current_usb_devices
+            interfaces = set(prev_usb_devices.values()).intersection(prev_usb_devices.values())
+            update_sharing_devices(interfaces)
+        await asyncio.sleep(timeout)
+
+
+def get_network_service(network_service_name: Optional[str]) -> NetworkService:
+    """ Retrieve the network service by name or return the default if none provided. """
+    network_preferences = NetworkPreferencePlist(INTERFACE_PREFERENCES)
+    if network_service_name is None:
+        network_service = get_default_route_network_service()
+        logger.info(
+            f'Network service name was not provided; using default: {network_service.interface.user_defined_name}'
+        )
+    else:
+        network_service = network_preferences.network_services.get_by_user_defined_name(network_service_name)
+        if network_service is None:
+            raise ValueError(f'Network service "{network_service_name}" not found')
+    return network_service
+
+
+def get_selected_devices(devices: Optional[tuple[str]]) -> list[str]:
+    """ Retrieve the selected USB devices. """
+    usb_devices = get_apple_usb_ethernet_interfaces()
+    if not usb_devices:
+        raise NoDeviceConnectedError()
+    if not devices:
+        udids = list(usb_devices.keys())
+        questions = [
+            inquirer3.Checkbox(
+                'Devices',
+                message='Choose devices',
+                choices=udids,
+                default=udids,
+            ),
+        ]
+        devices = inquirer3.prompt(questions)['Devices']
+    try:
+        return [usb_devices[x] for x in devices]
+    except KeyError as e:
+        raise DeviceNotFoundError(e.args[0])
+
+
 @click.group()
 def cli() -> None:
```

Comment:
Add context settings: `CONTEXT_SETTINGS = dict(help_option_names=['-h', '--help'], max_content_width=400)` so the help messages aren't cut short